src/main/java/com/fasterxml/jackson/databind/ObjectMapper.java

```diff
@@ -4789,6 +4789,24 @@ public void acceptJsonFormatVisitor(JavaType type, JsonFormatVisitorWrapper visi
         _serializerProvider(getSerializationConfig()).acceptJsonFormatVisitor(type, visitor);
     }
 
+    /*
+    /**********************************************************
+    /* Extended Public API: caches
+    /**********************************************************
+     */
+
+    /**
+     * Method that will clear all caches.
+     * This can be used to remove memory usage or to force re-construction cached entries after configuration changes.
+     * Can also be used to avoid class-loader memory leaks when reloading applications.
+     */
+    public void clearCaches() {
+        _rootDeserializers.clear();
+        getTypeFactory().clearCache();
+        getDeserializationContext().flushCachedDeserializers();
```

Comment:
This would work, although would be nice to see if there's any way around constructing throw-away context (if not, no worries).

But another question here: are `_rootDeserializers` separate from ones accessed here?

Reply:
I think there shouldn't be need to call both `_rootDeserializers.clear();` and `getDeserializationContext().flushCachedDeserializers();` -- `src/main/java/com/fasterxml/jackson/databind/cfg/CacheProvider.java` f.ex only has settings for 3 caches.

But I guess I can play with the test and see how things go. :)